src/com/facebook/buck/jvm/kotlin/KotlincToJarStepFactory.java

```diff
@@ -434,6 +445,24 @@ private String encodeOptions(Map<String, String> options) {
     }
   }
 
+  private String getFriendsPath(
+      SourcePathResolver sourcePathResolver, ImmutableList<SourcePath> friendPathsSourcePaths) {
+    if (friendPathsSourcePaths.isEmpty()) {
+      return "";
+    }
+
+    // https://youtrack.jetbrains.com/issue/KT-29933
+    ImmutableSortedSet<String> absoluteFriendPaths =
+        ImmutableSortedSet.copyOf(
+            friendPathsSourcePaths
+                .stream()
+                .map(path -> sourcePathResolver.getAbsolutePath(path).toString())
+                .collect(Collectors.toSet()));
+
+    return "-Xfriend-paths="
```

Comment:
Can this be detached from the arg? It would make adding it to the iml facet later much easier

```xml
<option name="friendPaths">
          <array>
            <option value="$MODULE_DIR$/../../../thrifty-compiler/build/classes/kotlin/main" />
            <option value="$MODULE_DIR$/../../../thrifty-compiler/build/classes/java/main" />
          </array>
        </option>
```